### PR TITLE
Filter backend

### PR DESCRIPTION
**Pull request by chenj47**
This PR involves an update for the backend to accommodate filtering-by-app and error visualization feature.
1. Add an app map to keep track of the sensors related to each app.
2. All sensors activity errors are now stored as csv values so that it can be shown by the visualizer.

### CODE REVIEW

File: sensorserviceutils/sensorserviceutils.go [hunk 1]
```diff
@@ -173,6 +173,11 @@ type parser struct {
 	parsingErrs []error
 	sensorErrs  []error
 
+	// apps is a map from UID to a sensorcheck map.
```
Comment (agarwal-v) [inline sensorserviceutils/sensorserviceutils.go]: sensorcheck - what do you mean by that? Do you mean
apps is a map from UID to active sensors.

File: sensorserviceutils/sensorserviceutils.go [hunk 2]
```diff
@@ -450,17 +486,39 @@ func (p parser) allSensorInfo() *sipb.AllSensorsInfo {
 	}
 	sort.Sort(allSensors)
 
+	allApp := make(appSlice, 0, len(p.apps))
```
Comment (agarwal-v) [inline sensorserviceutils/sensorserviceutils.go:490]: Is the TODO a tline 469 still a TODO?

File: pb/sensorsinfo_proto/sensorsinfo.proto [hunk 3]
```diff
@@ -81,8 +81,7 @@ message Sensor{
 message App{
     int32 UID = 1;
     string PackageName = 2;
-    repeated SubscriptionInfo Subscriptions = 3;
-    repeated ActiveConn ActiveConns = 4;
+    repeated string SensorActivities = 3;
```
Comment (agarwal-v) [inline pb/sensorsinfo_proto/sensorsinfo.proto:84]: What is the motivation behind this change?

**Review by agarwal-v — COMMENTED** [review]
Minor comments & questions

File: sensorserviceutils/sensorserviceutils.go (hunk 1 above)
Comment (chenj47) [reply sensorserviceutils/sensorserviceutils.go]: I think I will rephrase this comment.
apps is a map from uid to another map that indicates whether the app with this UID has a subscription event for the sensor. 
For example, if the inner map for UID=100 maps SensorNumber 3 to true,(p.apps[100][3] = true), it means that app with UID 100 has subscribed sensor number 3 in the history.

File: sensorserviceutils/sensorserviceutils.go (hunk 2 above)
Comment (chenj47) [reply sensorserviceutils/sensorserviceutils.go:490]: No, the feature has been implemented. I will update the comment.

File: pb/sensorsinfo_proto/sensorsinfo.proto (hunk 3 above)
Comment (chenj47) [reply pb/sensorsinfo_proto/sensorsinfo.proto:84]: The motivation is to avoid storing unnecessary information.
To implement the filtering-by-app feature, we just need to know what sensors have related activities to this app. Such information can be simply captured by recording the sensor name, rather than the actual events or active connections corresponding to the app. 
In this case, the filtering-by-app feature is fairly simple: we can do a first filtering to only keep the sensors whose name has shown in the list. Then only show events whos UID matches the app we are looking at.

**Review by agarwal-v — APPROVED** [review]
